src/components/CartItem/CartItem.jsx

```diff
@@ -0,0 +1,128 @@
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+import useDelete from 'hooks/shared/useDelete';
+import QuantityInput from 'components/QuantityInput';
+import comma from 'utils/comma';
+import Checkbox from 'components/Checkbox';
+import trash from 'assets/svg/trash.svg';
+
+import useGetCartList from 'hooks/useGetCartList';
+import { API_PATH } from 'constants/path';
+
+const CartItem = ({ item, onChangeEachCheckbox, checked }) => {
+  const { callDeleteApi } = useDelete(API_PATH.CART_LIST);
+  const [isChecked, setIsChecked] = useState(true);
+  const { getCartList } = useGetCartList();
+
+  useEffect(() => {
+    setIsChecked(checked);
+  }, [checked]);
+
+  const handleChangeCheckbox = () => {
+    setIsChecked(!isChecked);
+    onChangeEachCheckbox(item.id, isChecked);
+  };
+
+  const handleDeleteButton = async () => {
+    await callDeleteApi(item.id);
+    await getCartList();
```

Comment:
`getCartList` 가 `promise` 를 반환하나용?
```suggestion
    getCartList();
```

Reply:
`getCartList()`를 콘솔로그로 찍어보니까  pending인 promise가 반환됨을 확인했습니다. 
하지만 이 handleDeleteButton에서 getCartList가 마지막에 호출되니까 굳이 await할 필요는 없을 것 같습니다.
혹시 제가 잘못 이해했을까용..? 그렇다면, 질문해주신 내용 관련하여 더 자세한 설명 부탁드려도 괜찮을까요? 😄

Reply:
https://github.com/woowacourse/react-shopping-cart/pull/116/files#diff-0b2547716375601cff7f0470f07888023053a3bacc0b8c311868c40753d9a25fR7-R9 를 먼저 봐야할 것 같아요 :)
`getCartList`가 `async` 로 작성되어있으니까 promise 를 반환하는 건 맞는데, `getCartList` 내부를 보면 의도대로 동작할 진 의문이네요..! (await dispatch....?! await getCartList() 만 보면 동기로 동작할 것 같은데 사실은 아니..)

Reply:
https://github.com/woowacourse/react-shopping-cart/pull/116#discussion_r882355241

위에서 주신 리뷰와 동일한 맥락인 것 같아서 해당 답변 붙여드려요.
감사합니다!